pkgs/build-support/cc-wrapper/add-hardening.sh

```diff
@@ -1,32 +1,42 @@
-hardeningFlags=(fortify stackprotector pic strictoverflow format relro bindnow)
-# Intentionally word-split in case 'hardeningEnable' is defined in Nix.
-hardeningFlags+=(${hardeningEnable[@]})
+allHardeningFlags=(fortify stackprotector pie pic strictoverflow format relro bindnow)
 hardeningCFlags=()
 hardeningLDFlags=()
 
-declare -A hardeningDisableMap
+declare -A hardeningDisableMap=()
+declare -A hardeningEnableMap=()
 
-# Intentionally word-split in case 'hardeningDisable' is defined in Nix. The
+# Create table of unsupported flags for this toolchain.
+for flag in @hardening_unsupported_flags@; do
```

Comment:
Do we still need the disabled map if `mkDerivation` converts everything to a whitelist? I rather have this filter the whitelist.

Reply:
Sounds good, will do :+1: